src/modal/modal-container.component.ts

```diff
@@ -76,12 +77,18 @@ export class ModalContainerComponent implements OnInit, OnDestroy {
     }
   }
 
-  @HostListener('click', ['$event'])
-  onClick(event: MouseEvent): void {
+  @HostListener('mousedown', ['$event'])
+  onClickStarted(event: MouseEvent): void {
+    this.clickStartedInside = event.target !== this._element.nativeElement;
+  }
+
+  @HostListener('mouseup', ['$event'])
+  onClickStop(event: MouseEvent): void {
+    const clickedInside = this.clickStartedInside || event.target !== this._element.nativeElement;
```

Comment:
IMO should be
`const clickedOutside = !this.clickStartedInside || event.target !== this._element.nativeElement;`